google/cloud/pubsub/internal/subscriber_connection_impl_test.cc

```diff
@@ -409,6 +409,52 @@ TEST(SubscriberConnectionTest, Pull) {
   t.join();
 }
 
+TEST(SubscriberConnectionTest, PullReturnsNoMessage) {
```

Comment:
Consider a test that:

1) Sets the retry policy to tolerate at most N failures.
2) Returns empty messages N times.
3) The response should be an error of some kind.

Reply:
Added test